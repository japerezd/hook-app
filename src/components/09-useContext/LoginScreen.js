import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    
    // 1. obtener referencia al userContext
    const { setUser} = useContext(UserContext)
    // 2. setUser
    // 3. {id:1234, name: 'jorge'}
    const user = {
        id: 1234,
        name: 'Jorge'
    }
    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <button 
                className="btn btn-primary"
                onClick={() => setUser(user)}
            >Login</button>
        </div>
    )
}
