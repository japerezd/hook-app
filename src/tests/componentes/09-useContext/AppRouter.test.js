import { shallow } from 'enzyme';
import React from 'react';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas sobre <AppRouter />', () => {

    // Debe hacerse con mount
    const wrapper = shallow(
    <UserContext.Provider value={{
        id: 1,
        name: 'Beto'
    }}>
        <AppRouter />
    </UserContext.Provider>
    );
    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    

})
