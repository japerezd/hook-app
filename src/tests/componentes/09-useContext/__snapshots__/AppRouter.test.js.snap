// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <AppRouter /> debe mostrarse correctamente 1`] = `<AppRouter />`;
